### PR TITLE
Rename "soft ack" to echo

### PR DESCRIPTION
**Pull request by mfalkvidd**
The ack parameter is often confused with what might be called
"hardware ack"
(see https://forum.mysensors.org/post/34263 for one of many
lengthy discussions)

This change should make it harder to confuse echo with the
hardware ack, and reflects more accurately what will actually
happen.

This fixes https://github.com/mysensors/MySensors/issues/1103

I hope I have found all places where the naming needs to be
changed, and that I haven't inadvertently renamed any of the
*real* ack stuff.

### CODE REVIEW

File: core/MySigning.cpp [hunk 1]
```diff
@@ -373,7 +373,7 @@ static bool skipSign(MyMessage &msg)
 		ret = true;
 	}
 	if (ret) {
-		SIGN_DEBUG(PSTR("SGN:SKP:%s CMD=%" PRIu8 ",TYPE=%" PRIu8 "\n"), mGetAck(msg) ? "ACK" : "MSG",
+		SIGN_DEBUG(PSTR("SGN:SKP:%s CMD=%" PRIu8 ",TYPE=%" PRIu8 "\n"), mGetEcho(msg) ? "ACK" : "MSG",
```
Comment (tekka007) [inline core/MySigning.cpp]: ACK => ECHO

File: core/MyMessage.h [hunk 2]
```diff
@@ -247,36 +247,36 @@ typedef enum {
...
 #define mGetLength(_message) ((uint8_t)BF_GET(_message.version_length, 3, 5)) //!< Get length field
 
-#define mSetCommand(_message,_command) BF_SET(_message.command_ack_payload, _command, 0, 3) //!< Set command field
-#define mGetCommand(_message) ((uint8_t)BF_GET(_message.command_ack_payload, 0, 3)) //!< Get command field
+#define mSetCommand(_message,_command) BF_SET(_message.command_echo_payload, _command, 0, 3) //!< Set command field
+#define mGetCommand(_message) ((uint8_t)BF_GET(_message.command_echo_payload, 0, 3)) //!< Get command field
 
-#define mSetRequestAck(_message,_rack) BF_SET(_message.command_ack_payload, _rack, 3, 1) //!< Set ack-request field
-#define mGetRequestAck(_message) ((bool)BF_GET(_message.command_ack_payload, 3, 1)) //!< Get  ack-request field
+#define mSetRequestEcho(_message,_rack) BF_SET(_message.command_echo_payload, _rack, 3, 1) //!< Set ack-request field
```
Comment (tekka007) [inline core/MyMessage.h]: doxygen comment says ack

File: core/MyMessage.h [hunk 3]
```diff
@@ -247,36 +247,36 @@ typedef enum {
...
+#define mGetRequestEcho(_message) ((bool)BF_GET(_message.command_echo_payload, 3, 1)) //!< Get  ack-request field
```
Comment (tekka007) [inline core/MyMessage.h]: doxygen comment says ack

File: core/MyMessage.h [hunk 4]
```diff
@@ -247,36 +247,36 @@ typedef enum {
...
 
-#define mSetAck(_message,_ackMsg) BF_SET(_message.command_ack_payload, _ackMsg, 4, 1) //!< Set ack field
-#define mGetAck(_message) ((bool)BF_GET(_message.command_ack_payload, 4, 1)) //!< Get ack field
+#define mSetEcho(_message,_ackMsg) BF_SET(_message.command_echo_payload, _ackMsg, 4, 1) //!< Set ack field
```
Comment (tekka007) [inline core/MyMessage.h]: doxygen comment says ack

File: core/MyMessage.h [hunk 5]
```diff
@@ -247,36 +247,36 @@ typedef enum {
...
+#define mGetEcho(_message) ((bool)BF_GET(_message.command_echo_payload, 4, 1)) //!< Get ack field
```
Comment (tekka007) [inline core/MyMessage.h]: doxygen comment says ack

File: core/MyOTALogging.cpp [hunk 6]
```diff
@@ -22,7 +22,7 @@
 // global variables
 static bool inOTALog = false;
 
-void OTALog(uint8_t logNode, bool enableAck, const char *fmt, ... )
+void OTALog(uint8_t logNode, bool enableEcho, const char *fmt, ... )
```
Comment (tekka007) [inline core/MyOTALogging.cpp]: for consistency, better use `echo`

header file also needs to be updated:
https://github.com/mysensors/MySensors/blob/development/core/MyOTALogging.h#L46-L54

**Comment by mfalkvidd** [comment]
Thanks for the feedback.

I'll need to dig deeper to understand if MY_DEBUG_OTA_DISABLE_ACK is "hardware ack" or "software ack". Same goes for core/MyMessage.h:/// @brief Type of sensor data (for set/req/ack messages)


Not sure when I'll have some time, might be a few days. Sorry for the delay.

**Comment by tekka007** [comment]
> Thanks for the feedback.
> 
> I'll need to dig deeper to understand if MY_DEBUG_OTA_DISABLE_ACK is "hardware ack" or "software ack"

The description of this directive is misleading:
https://github.com/mysensors/MySensors/blob/development/MyConfig.h#L2086-L2090
but it is a SW ack (=echo): https://github.com/mysensors/MySensors/blob/development/MyConfig.h#L2086-L2090

**Comment by tekka007** [comment]
> Same goes for core/MyMessage.h:/// @brief Type of sensor data (for set/req/ack messages)

refers to v_type of ack message ==> SW ack (=echo)

File: core/MyOTALogging.cpp (hunk 6 above)
Comment (tekka007) [reply core/MyOTALogging.cpp]: see initial comment

File: core/MyOTALogging.h [hunk 7]
```diff
@@ -47,11 +47,11 @@
  * new debug message line starts. Incomplete messages are ending with '...'
  *
  * @param logNode    Destination node ID
- * @param enableAck  Enable or disable ACK flag
+ * @param enableEcho Enable or disable echo flag
  * @param fmt        printf format string
  * @param ...        arguments
  */
-void OTALog(uint8_t logNode, bool enableAck, const char *fmt, ... );
+void OTALog(uint8_t logNode, bool enableEcho, const char *fmt, ... );
```
Comment (tekka007) [inline core/MyOTALogging.h]: for consistency, use `echo` param

File: examples/LogOTANode/LogOTANode.ino [hunk 8]
```diff
@@ -33,8 +33,8 @@
 // Allow sending logs without MY_DEBUG_OTA enabled
 #define MY_OTA_LOG_SENDER_FEATURE
 
-// Disable ACK for debug messages
-//#define MY_DEBUG_OTA_DISABLE_ACK
+// Disable echoing of debug messages
+//#define MY_DEBUG_OTA_DISABLE_ECHO
```
Comment (tekka007) [inline examples/LogOTANode/LogOTANode.ino]: maybe adding a #warning if the old directive is used

File: MyConfig.h [hunk 9]
```diff
@@ -85,14 +85,15 @@
 //#define MY_DEBUG_OTA (0)
 
 /**
- * @def MY_DEBUG_OTA_DISABLE_ACK
- * @brief Define MY_DEBUG_OTA_DISABLE_ACK to send messages with no ACK flag.
+ * @def MY_DEBUG_OTA_DISABLE_ECHO
+ * @brief Define MY_DEBUG_OTA_DISABLE_ECHO to send messages without requesting the
+ * estination to echo the message.
```
Comment (tekka007) [inline MyConfig.h]: typo: **d**estination

**Pull request merged**